packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -811,42 +814,45 @@ class _ModalScopeState<T> extends State<_ModalScope<T>> {
                     controller: primaryScrollController,
                     child: FocusScope(
                       node: focusScopeNode, // immutable
-                      child: RepaintBoundary(
-                        child: AnimatedBuilder(
-                          animation: _listenable, // immutable
-                          builder: (BuildContext context, Widget? child) {
-                            return widget.route.buildTransitions(
-                              context,
-                              widget.route.animation!,
-                              widget.route.secondaryAnimation!,
-                              // This additional AnimatedBuilder is include because if the
-                              // value of the userGestureInProgressNotifier changes, it's
-                              // only necessary to rebuild the IgnorePointer widget and set
-                              // the focus node's ability to focus.
-                              AnimatedBuilder(
-                                animation: widget.route.navigator?.userGestureInProgressNotifier ?? ValueNotifier<bool>(false),
-                                builder: (BuildContext context, Widget? child) {
-                                  final bool ignoreEvents = _shouldIgnoreFocusRequest;
-                                  focusScopeNode.canRequestFocus = !ignoreEvents;
-                                  return IgnorePointer(
-                                    ignoring: ignoreEvents,
-                                    child: child,
+                      child: _FocusTrap(
+                        focusScopeNode: focusScopeNode,
```

Comment:
The same focusnode get passed into two widgets. not sure if this is a concern cc @gspencergoog 

If it is, we can avoid this by merging the FocusScope and _focusTrap into one widget.

Reply:
Usually sharing a `FocusNode` (or `FocusScopeNode`) between two widgets is a concern because they'll fight over the attachment.  If the focus node is _only_ used to call `requestFocus()`, or something else that doesn't require attachment, in one of the widgets, then it's technically OK.

Basically, the rule is: Don't pass the same focus node to two widgets that will attach it to the focus tree. Only one widget can "own" it.

You might also have lifetime issues when it comes to disposing them. Again, only one may dispose of the focus node, and so the widget that it is passed to that doesn't own it needs to have a shorter lifetime than the one that does own it.

Reply:
I think what I'm doing here is safe, no stack traces or bugs so far. What would you like me to change about the current structure?

Reply:
I don't have a good suggestion neither, if @gspencergoog is ok with the current structure, I am fine with it as well

Reply:
I think I'm OK with the current structure: the widget using the node is a private class, so we have control over its lifetime, and it doesn't appear to be doing anything that would attach the node to the focus tree.